packages/babel-parser/src/parser/statement.ts

```diff
@@ -580,6 +580,14 @@ export default abstract class StatementParser extends ExpressionParser {
         expr = this.parseExpression();
         this.expect(tt.parenR);
         expr = this.wrapParenthesis(startPos, startLoc, expr);
+
+        if (
+          this.getPluginOption("decorators", "allowCallParenthesized") !== false
+        ) {
+          node.expression = this.parseMaybeDecoratorArguments(expr);
+        } else {
+          node.expression = expr;
+        }
```

Comment:
Can we always parse decorator arguments and throw a recoverable error when `allowCallParenthesized` is `false`?